### PR TITLE
ci: added helm-test chart workflow (TRG 5.07)

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

Adds the helm test workflow according to TRG 5.07. To clarify, if helm upgrade between charts is applicable to us.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by tom-rm-meyer-ISST** [comment]
Hey @almadigabor, could you do a review and tell me why it fails due to kind version 1.27.3?

File: .github/workflows/helm-test.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+name: Lint and Test Charts
+
+on:
+    pull_request:
+    workflow_dispatch:
+        inputs:
+            node_image:
+                description: 'kindest/node image for k8s kind cluster'
+                # k8s version from 3.1 release as default
+                default: 'kindest/node:1.27.3'
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:28]: Issue due to a typo in the image tag (lost 'v')  hence unable to download it.

Use: kindest/node:v1.27.3

File: .github/workflows/helm-test.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              with:
+                  fetch-depth: 0
+
+            - name: Kubernetes KinD Cluster
+              uses: container-tools/kind-action@7075d1458484493c6a92d4604cb27b87de0f8107 #v2.2
+              with:
+                  # upgrade version, default (v0.17.0) uses node image v1.21.1 and doesn't work with more recent node image versions
+                  version: v0.20.0
+                  # default value for event_name != workflow_dispatch
+                  node_image: ${{ github.event.inputs.node_image || 'kindest/node:1.27.3' }}
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:52]: Same as above. Typo issue in the image name.

**Review by tomaszbarwicki — CHANGES_REQUESTED** [review]

File: .github/workflows/helm-test.yml (hunk 1 above)
Comment (tom-rm-meyer-ISST) [reply .github/workflows/helm-test.yml]: Thanks for pointing out! fixed

**Comment by tom-rm-meyer-ISST** [comment]
@tomaszbarwicki, linting progresses but the dockerfile can't be read in this [run](https://github.com/eclipse-tractusx/puris/actions/runs/7409214943/job/20159030807?pr=145).
```
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2079917770/Dockerfile: no such file or directory
Error: buildx failed with: ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2079917770/Dockerfile: no such file or directory
```

Do you got an idea why?

**Comment by almadigabor** [comment]
> @tomaszbarwicki, linting progresses but the dockerfile can't be read in this [run](https://github.com/eclipse-tractusx/puris/actions/runs/7409214943/job/20159030807?pr=145).
> 
> ```
> ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2079917770/Dockerfile: no such file or directory
> Error: buildx failed with: ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2079917770/Dockerfile: no such file or directory
> ```
> 
> Do you got an idea why?

The current issue happens because the action does not point to the Dockerfile's path using the `context` variable. You have your Dockerfiles in the frontend and backend directory. Based on your setup you should build both by duplicating the build step and create a frontend and backend image as well. Then in the end make sure to set the proper registry, app name and tag in the chart install step. You can find a good example [here](https://github.com/eclipse-tractusx/traceability-foss/blob/9d49d92ffc6a7121d6e0a9cc35dee22fc943331b/.github/workflows/helm-test.yaml) building frontend and backend images and then using them with the chart at the end.

https://github.com/eclipse-tractusx/puris/pull/145/files#diff-33f5fdd3507841f6bb8f87595ecc5669a0397021de950482ca4e8b68c25cfdb6R57

**Comment by tom-rm-meyer-ISST** [comment]
Hey @almadigabor, needed some free space to have a look into it again. It now works. Thanks for your support. Could you please do the final review and accept it?

**Review by almadigabor — APPROVED** [review]
The workflow looks good now, although the actual [chart-testing (install) step did not run](https://github.com/eclipse-tractusx/puris/actions/runs/7728019500/job/21067860716?pr=145) as no chart changes were detected. Anyways I'll approve this PR.

**Comment by almadigabor** [comment]
Hey @tomaszbarwicki! Looks like you requested some changes that are outdated now. Can you re-review this PR?

**Review by tomaszbarwicki — APPROVED** [review]
LGTM

**Pull request merged**